File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/PassingWaypoint.java
package net.cyclestreets.liveride;

import net.cyclestreets.routing.Journey;

import org.osmdroid.util.GeoPoint;

final class PassingWaypoint extends LiveRideState
{
  PassingWaypoint(final LiveRideState previous) {
    super(previous);
    notify("Passing waypoint");
  }

  @Override
  public LiveRideState update(Journey journey, GeoPoint whereIam, int accuracy) {
    return new AdvanceToSegment(this, journey);
  }

  @Override
  public boolean isStopped() { return false; }
  @Override
  public boolean arePedalling() { return true; }
}

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/PassingWaypoint.kt
package net.cyclestreets.liveride

import net.cyclestreets.routing.Journey
import org.osmdroid.util.GeoPoint

internal class PassingWaypoint(previous: LiveRideState?) : LiveRideState(previous!!) {

    init {
        notify("Passing waypoint")
    }

    override fun update(journey: Journey, myLocation: GeoPoint, accuracy: Int): LiveRideState {
        return AdvanceToSegment(this, journey)
    }

    override fun isStopped(): Boolean { return false }
    override fun arePedalling(): Boolean { return true }
}
